app/src/main/java/com/stevesoltys/seedvault/transport/backup/PackageService.kt

```diff
@@ -157,9 +157,20 @@ internal fun PackageInfo.isSystemApp(): Boolean {
     return applicationInfo.flags and FLAG_SYSTEM != 0
 }
 
-internal fun PackageInfo.allowsBackup(): Boolean {
+internal fun PackageInfo.allowsBackup(context: Context, userId: Int): Boolean {
     if (packageName == MAGIC_PACKAGE_MANAGER || applicationInfo == null) return false
-    return applicationInfo.flags and FLAG_ALLOW_BACKUP != 0
+
+    // This must be kept in sync with frameworks/base/services/backup/java/com/android/server/backup/utils/BackupEligibilityRules.java
+    val isSystemApp = applicationInfo.flags and FLAG_SYSTEM != 0
```

Comment:
Separate discussion but do we want to change this too?

I believe it may need an OS change as well.

I can't think of many system apps this would be useful for, except perhaps Chromium (if it even works)

Otherwise we can easily just directly change the system app, so I think probably best to keep things like they were for system apps (aka what this commit currently has)